src/qt/bitcoingui.cpp

```diff
@@ -1343,8 +1343,10 @@ void BitcoinGUI::setNumBlocks(int count, const QDateTime& blockDate, const QStri
     // Disabling macOS App Nap on initial sync, disk, reindex operations and mixing.
     bool disableAppNap = !m_node.masternodeSync().isSynced();
 #ifdef ENABLE_WALLET
-    for (const auto& wallet : m_node.walletClient().getWallets()) {
-        disableAppNap |= wallet->coinJoin().isMixing();
+    if (walletFrame != nullptr) {
```

Comment:
I'm confused, why are you checking walletFrame here? Is there not a better way to tell if a wallet is enabled or not?

Reply:
That's how it works here in `bitcoingui.cpp` since it was originally introduced in bitcoin#3392

Reply:
I guess it's used all over the place instead of `WalletModel::isWalletEnabled()` for example cause it's the most lightweight way to do it (avoids checking "-disablewallet" all the time).

Reply:
I don't really know if I like that... I'd prefer if it was clearer (the method you mentioned)

But if you strongly disagree then we can move forward with this pr.